src/hip_memory.cpp

```diff
@@ -1969,14 +1970,27 @@ hipError_t hipMemGetInfo(size_t* free, size_t* total) {
         }
 
         if (free) {
-            // TODO - replace with kernel-level for reporting free memory:
-            size_t deviceMemSize, hostMemSize, userMemSize;
-            hc::am_memtracker_sizeinfo(device->_acc, &deviceMemSize, &hostMemSize, &userMemSize);
-
-            *free = device->_props.totalGlobalMem - deviceMemSize;
-
-            // Deduct the amount of memory from the free memory reported from the system
-            if (HIP_HIDDEN_FREE_MEM) *free -= (size_t)HIP_HIDDEN_FREE_MEM * 1024 * 1024;
+            hsa_agent_t* agent = static_cast<hsa_agent_t*>(device->_acc.get_hsa_agent());
```

Comment:
Looks like implementation assumes that 'device' will be not null if 'context' exists, can you please confirm that?

Reply:
Yes, as we cannot create a context without having an associated device.